### PR TITLE
add more logging to vmi.wait_until_running()

### PR DESCRIPTION
**Pull request by vsibirsk**
##### Short description:
Add more logging to VirtualMachineInstance.wait_until_running()

### CODE REVIEW

File: ocp_resources/virtual_machine_instance.py [hunk 1]
```diff
@@ -115,6 +115,7 @@ def wait_until_running(self, timeout=TIMEOUT_4MINUTES, logs=True, stop_status=No
             TimeoutExpiredError: If VMI failed to run.
         """
         try:
+            self.logger.info(f"VMI status before wait: {self.instance.status.phase}")
```
Comment (myakove) [inline ocp_resources/virtual_machine_instance.py:118]: You should also log the vmi name

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/virtual_machine_instance.py [hunk 2]
```diff
@@ -124,6 +125,9 @@ def wait_until_running(self, timeout=TIMEOUT_4MINUTES, logs=True, stop_status=No
```
Comment (rnetser) [inline ocp_resources/virtual_machine_instance.py:125]: if vmi has a vm please also log vm status

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/virtual_machine_instance.py (hunk 1 above)
Comment (vsibirsk) [reply ocp_resources/virtual_machine_instance.py]: done

File: ocp_resources/virtual_machine_instance.py (hunk 2 above)
Comment (vsibirsk) [reply ocp_resources/virtual_machine_instance.py:127]: vmi doesn't have vm attribute. will need to add new property for it

Comment (rnetser) [reply ocp_resources/virtual_machine_instance.py:127]: VirtualMachine.get(name=self.name, namespace=self.namespace) ....

Comment (vsibirsk) [reply ocp_resources/virtual_machine_instance.py:135]: done

Comment (rnetser) [reply ocp_resources/virtual_machine_instance.py:135]: the comment was for the exception part :)  leave the above log and add also right after `except TimeoutExpiredError`

File: ocp_resources/virtual_machine_instance.py [hunk 3]
```diff
@@ -102,6 +103,13 @@ def virt_handler_pod(self):
 
         raise ResourceNotFoundError
 
+    @property
+    def vm(self):
+        vms = list(VirtualMachine.get(name=self.name, namespace=self.namespace))
```
Comment (myakove) [inline ocp_resources/virtual_machine_instance.py:108]:
```
vm = VirtualMachine(name=self.name, namespace=self.namespace)
if vm.exists:
    return vm
raise ResourceNotFoundError(.......error......)
````

File: ocp_resources/virtual_machine_instance.py [hunk 4]
```diff
@@ -124,6 +138,9 @@ def wait_until_running(self, timeout=TIMEOUT_4MINUTES, logs=True, stop_status=No
 
             virt_pod = self.virt_launcher_pod
             if virt_pod:
+                self.logger.error(
+                    f"Status of virt-launcher pod {virt_pod.name}: {virt_pod.status}"
```
Comment (myakove) [inline ocp_resources/virtual_machine_instance.py:142]: Please also log here vm and vmi names
can be something like:
`Status of virt-launcher pod {virt_pod.name}: {virt_pod.status} [VM: ......, VMI: .....]`

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/virtual_machine_instance.py (hunk 3 above)
Comment (rnetser) [reply ocp_resources/virtual_machine_instance.py:108]: raise is not needed as vmi can exist without a vm

Comment (rnetser) [reply ocp_resources/virtual_machine_instance.py:108]: get if from the vmi
metadata.ownerReferences.....

```
  ownerReferences:
  - apiVersion: kubevirt.io/v1
    blockOwnerDeletion: true
    controller: true
    kind: VirtualMachine
    name: fedora-relieved-beetle
    uid: c7435302-d1da-4d78-abfd-d9b7ac384544

```

**Review by rnetser — DISMISSED** [review]

File: ocp_resources/virtual_machine_instance.py (hunk 3 above)
Comment (rnetser) [reply ocp_resources/virtual_machine_instance.py:108]: please remove (and also the log about VM below)

File: ocp_resources/virtual_machine_instance.py (hunk 4 above)
Comment (vsibirsk) [reply ocp_resources/virtual_machine_instance.py:131]: added vmi name + status in exception block

File: ocp_resources/virtual_machine_instance.py (hunk 3 above)
Comment (vsibirsk) [reply ocp_resources/virtual_machine_instance.py]: removed

**Review by myakove — APPROVED** [review]

**Review by myakove — APPROVED** [review]

**Pull request merged**